docs/en/migrations/snowflake.md

```diff
@@ -0,0 +1,110 @@
+
+## 1. Exporting data from Snowflake
+
+<img src={require('./images/migrate_snowflake_clickhouse.png').default} class="image" alt="Migrating from Snowflake to ClickHouse" style={{width: '600px', marginBottom: '20px', textAlign: 'left'}}/>
+
+Exporting data from Snowflake requires the use of an external stage, as shown in the diagram above.
+
+Let's say we want to export a Snowflake table with the following schema:
+
+```sql
+CREATE TABLE MYDATASET (
+   timestamp TIMESTAMP,
+   some_text varchar,
+   some_file OBJECT,
+   complex_data VARIANT,
+) DATA_RETENTION_TIME_IN_DAYS = 0;
+```
+
+To move this table's data to a ClickHouse database, we first need to copy this data to an external stage. When copying data, we recommend Parquet as the intermittent format as it allows type information to be shared, preserves precision, compresses well, and natively supports nested structures common in analytics.
+
+In the example below, we create a named file format in Snowflake to represent Parquet and the desired file options. We then specify which bucket will contain our copied dataset. Finally, we copy the dataset to the bucket.
+
+```sql
+CREATE FILE FORMAT my_parquet_format TYPE = parquet;
+
+-- Create the external stage that specifies the S3 bucket to copy into
+CREATE OR REPLACE STAGE external_stage 
+URL='s3://mybucket/mydataset'
+CREDENTIALS=(AWS_KEY_ID='<key>' AWS_SECRET_KEY='<secret>')
+FILE_FORMAT = my_parquet_format;
+
+-- Apply "mydataset" prefix to all files and specify a max file size of 150mb
+-- The `header=true` parameter is required to get column names
+COPY INTO @external_stage/mydataset from mydataset max_file_size=157286400 header=true;
+```
+
+For a dataset around 5TB of data with a maximum file size of 150MB, and using a 2X-Large Snowflake warehouse located in the same AWS `us-east-1` region, copying data to the S3 bucket will take around 30 minutes.
+
+## 2. Importing to ClickHouse
+
+Once the data is staged in intermediary object storage, ClickHouse functions such as the [s3 table function](/docs/en/sql-reference/table-functions/s3) can be used to insert the data into a table, as shown below.
+
+This example uses the [s3 table function](/docs/en/sql-reference/table-functions/s3) for AWS S3, but the [gcs table function](/docs/en/sql-reference/table-functions/gcs) can be used for Google Cloud Storage and the [azureBlobStorage table function](/docs/en/sql-reference/table-functions/azureBlobStorage) can be used for Azure Blob Storage.
+
+Assuming the following table target schema:
+
+```sql
+CREATE TABLE default.mydataset
+(
+	`timestamp` DateTime64(6),
+	`some_text` String,
+	`some_file` Tuple(filename String, version String),
+	`complex_data` Tuple(name String, description String),
+)
+ENGINE = MergeTree
+ORDER BY (timestamp)
+```
+
+We can then use the `INSERT INTO SELECT` command to insert the data from S3 into a ClickHouse table:
+
+```sql
+INSERT INTO mydataset
+SELECT
+	timestamp,
+	some_text,
+	JSONExtract(
```

Comment:
last time i checked they do export OBJECT as JSON string..this is subject to change. i raised a question and they planned to possibly change. Maybe add note?